DEVELOPMENT.md

```diff
@@ -0,0 +1,96 @@
+
+##Aurora features
+Handlers for Aurora features are defined in `no.skatteetaten.aurora.boober.feature`.
+
+### Features
+Features must extend the interface [Feature](./src/main/kotlin/no/skatteetaten/aurora/boober/feature/Feature.kt).
+The interface include documentation for the functions that needs to be implemented.
+
+The implemented functions are used in [AuroraDeploymentContext](./src/main/kotlin/no/skatteetaten/aurora/boober/model/AuroraDeploymentContext.kt)
+used by [AuroraDeploymentContextService](./src/main/kotlin/no/skatteetaten/aurora/boober/service/AuroraDeploymentContextService.kt)
+which in turn is used by [DeployFacade](./src/main/kotlin/no/skatteetaten/aurora/boober/facade/DeployFacade.kt). 
+
+
+The interface provides two functions for generating resources: `generate` and `generateSequentially`.
+A feature will normally only need to implement the `generate` function which will be executed in parallel.
+The `generateSequentially` function is intended for resources where parallel generation may result in duplicate resources (for example shared databases [DatabaseFeature](./src/main/kotlin/no/skatteetaten/aurora/boober/feature/DatabaseFeature.kt)). 
+
+#### Handlers
+The handlers function returns a set of [AuroraConfigFieldHandler](./src/main/kotlin/no/skatteetaten/aurora/boober/model/AuroraConfigFieldHandler.kt)
+which provides handling of a given pointer with default value and validation.
+- Pre-defined validation logic can be found in [jsonNodeUtils.kt](./src/main/kotlin/no/skatteetaten/aurora/boober/utils/jsonNodeUtils.kt).
+- Validation severity can be set with the validationSeverity parameter (default value is [ErrorType.ILLEGAL](./src/main/kotlin/no/skatteetaten/aurora/boober/model/errors.kt)).
+- `ErrorType.WARNING` can be used when the validation should warn about configuration but allow deploying.
+
+#### AuroraDeploymentSpec
+[AuroraDeploymentSpec](./src/main/kotlin/no/skatteetaten/aurora/boober/model/AuroraDeploymentSpec.kt) is a class passed to `validate`, `generate` and `generateSequentially`.
+The class contains the fields for a given deployment and helper functions for extracting fields for validation and generation purposes.
+It is recommended to look at how existing features makes use of the class.
```

Comment:
Det er vel verdt å merke seg at det finnes en header spec som man får i handlers funksjonen. 
Ellers får man headerspec + spec for alle handlers featuren har definert.

Reply:
jeg la på litt mer detaljer om dette